src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -273,14 +271,8 @@ class LinearSVM
   ElemType Train(const MatType& data,
                  const arma::Row<size_t>& labels,
                  const size_t numClasses,
-                 const double lambda);
-
-  template<typename MatType>
-  ElemType Train(const MatType& data,
-                 const arma::Row<size_t>& labels,
-                 const size_t numClasses,
-                 const double lambda,
-                 const double delta);
+                 const std::optional<double> lambda = std::nullopt,
+                 const std::optional<double> delta = std::nullopt);
```

Comment:
As far as I can tell it is also possible to combine this with the version that has callbacks below; I tested it locally.  The signature is this:

```c++
  template<typename MatType,
           typename... CallbackTypes,
           typename = typename std::enable_if<IsEnsCallbackTypes<
               CallbackTypes...
           >::value>::type>
  ElemType Train(const MatType& data,
                 const arma::Row<size_t>& labels,
                 const size_t numClasses,
                 const std::optional<double> lambda = std::nullopt,
                 const std::optional<double> delta = std::nullopt,
                 const std::optional<bool> fitIntercept = std::nullopt,
                 CallbackTypes&&... callbacks);
```

Did you encounter any problems when trying that?  It could allow us to remove another overload.

Reply:
When I tried the code above, I had no compilation issues in the tests.  Do you have a particular test case that fails to compile?  I can try and debug it.